test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -196,6 +197,8 @@ public void run() {
         }
     };
 
+    private final IdentityHashMap<Method, Method> tcclMethodCache = new IdentityHashMap<>();
```

Comment:
This needs to be cleared at the end of the run, otherwise continuous testing will leak ClassLoaders.

Reply:
Ok, where is the best place to do that? Here in `afterAll`? https://github.com/quarkusio/quarkus/blob/2.2.0.Final/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L1134

Reply:
I think here: https://github.com/stuartwdouglas/quarkus/blob/78226e2971feea96fc2a63362529413b957aa120/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L277

Reply:
Please do it just before the `GroovyCacheCleaner` call given we have little control on this one and we want to make sure this cache is reset.

Reply:
Done. I must say though it's a bit surprising that this non-static cache causes cl leaks, given that it should be GC-able as soon as the next test class is executed (= new instance of `QuarkusTestExtension` is created).
But then again I know little about the internals of continuous testing and I guess it's always safer to clear things eagerly, if possible.